src/libstd/collections/hash/map.rs

```diff
@@ -1378,13 +1378,14 @@ impl<K, V, S> Hash for HashMap<K, V, S>
 {
     fn hash<H: Hasher>(&self, hasher: &mut H) {
         // We must preserve:  x == y -> hash(x) == hash(y).
-        // HashMaps have no order, so we must use a commutative operation
-        // (.wrapping_add) so that the order does not matter.
+        // HashMaps have no order, so we must combine the elements with an
+        // associative and commutative operation • so that the order does not
+        // matter. In other words, (u64, •, 0) must form a commutative monoid.
```

Comment:
In theory, it would be sufficient to use a pointed commutative semigroup, as you're not making use of the identity at all here, but I think there's a limit to technicalities 😄This looks fine to me!

Reply:
Haha =P The fold uses the identity element 0 though.

Reply:
The fold just needs an initial (pointed) element (which doesn't need to be special in any way), right? You've picked the identity, but in theory you could have chosen any other `u64`.

Reply:
Oh yes, that is true =P /nerdsniped